### PR TITLE
Use dask.delayed instead of the distributed client

### PR DESCRIPTION
**Pull request by leouieda**
Started work on using dask.delayed instead of the futures interface
(client.submit). It's easier and allows building the entire graph lazily
before executing. Still need tests and to port the SplineCV code to
this. Will deprecate the client argument and remove in 2.0.0.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst` and `verde/__init__.py`.
- [ ] Write detailed docstrings for all functions/classes/methods.
- [ ] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.

### CODE REVIEW

**Comment by leouieda** [comment]
Having Dask be optional is causing headaches:

1. Can't use it in the documentation AND have builds that don't install Dask to test that it's really optinal. Since we build the docs always, there is no easy way to have both.
2. Need to wrap all calls to dask in functions that raise exceptions when it's not installed. This increases the amount of tests for each of these functions and is a bit tedious to write.

Dask is a really lightweight dependency (pure Python with minimal requirements) and I would like to add more support for it in the future (to enable parallel fitting and larger than RAM Jacobians). 

**I'm thinking we should just adopt Dask as a dependency. Scikit-learn is a much heavier dependency and we already have it. This is nothing compared to GDAL and the geospatial stack.** 

@santisoler @jessepisel any thoughts/objections on this?

**Comment by santisoler** [comment]
I'm seeing Dask in the future of Verde, specially for solving problems that include larger-than-memory arrays. So I think we could add it as a dependency right away and save the headaches of writing annoying lines on test functions and struggling with building the docs.
It's lightweight, so I don't see why we shouldn't add it as a dependency.

**Comment by leouieda** [comment]
:+1: alright, I'll try to get this one finished soon. Thanks, @santisoler

**Comment by jessepisel** [comment]
I agree with @santisoler that Dask should be a dependency. It will be easier to deprecate the futures interface and not have to write new tests for both options. This will be nice to have for big grids that don't fit in memory @leouieda !

**Comment by leouieda** [comment]
Thanks for the input @jessepisel and @santisoler :+1: 

> This will be nice to have for big grids that don't fit in memory @leouieda !

I'm still struggling with ways of doing this. The dask-ml package has some optimization methods for linear problems that we could use to run the least-squares fit. But it's not stable enough to be a dependency. Right now, solving the system with the linear algebra is dask is bad because there are a lot of limitations (need square chunks, for one). That's where we should start looking at [pylops](https://github.com/equinor/pylops). Matteo is working on a distributed version of that using dask. I'm eager to play around with it and see if we can make it work.

**Comment by leouieda** [comment]
Another option is to break up the fit into windows and run each window separately (using dask).

**Comment by leouieda** [comment]
OK, I added the delayed argument to the tutorials as well and made Dask a dependency. @santisoler and/or @jessepisel could you please review this PR?

File: tutorials/model_evaluation.py [hunk 1]
```diff
@@ -184,7 +184,53 @@
...
+print("Delayed mean:", mean_score)
+
+########################################################################################
+# To run the scheduled computations and get the mean score, use
+# :func:`dask.compute` or ``.compute()``. Dask will automatically execute
+# things in parallel.
+
+mean_score = mean_score.compute()
+print("Mean score:", mean_score)
+
```
[the remark this answers is not in the document]
Comment (leouieda) [reply tutorials/model_evaluation.py]: :+1: thanks, Jesse!

**Comment by leouieda** [comment]
Thanks for the review @jessepisel! 

>  is there any reason why you changed the minimum distances to fit by such a large value? Is it ensuring that the 1e-7 is returned for the test? That's all I noticed for the tests.

That is exactly it. I just wanted to make it easier for the CV by giving it obviously bad values and only 1 good combination. This will avoid future headaches due to floating point round-off.

**Pull request merged**